java-checks/src/test/files/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -153,4 +159,32 @@ void unknownWarningCombinedWithKnown(List list, int unused) { // Compliant
   }
 }
 
+class StrutsAction extends Action {
+  void foo(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response, String s) { // Compliant
+    System.out.println(s); 
+  }
+  
+  void bar(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response, String unused) { // Noncompliant {{Remove this unused method parameter "unused".}}
+    System.out.println(""); 
+  }
+  
+  void qix(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) { // Compliant
+    System.out.println(""); 
+  }
+  
+  void qiz(ActionMapping mapping, ActionForm form) { // Compliant
+    System.out.println(""); 
+  }
+}
+
+class StrutsAction2 extends BaseAction {
+  void foo(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response, String unused) { // Noncompliant {{Remove this unused method parameter "unused".}}
+    System.out.println(""); 
+  }
+  
+  void bar(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) { // Compliant
```

Comment:
In order to completely cover you cases, add a method in a 3rd class which does not extends Action, having an unused parameter being from one of the struts types. This should raise an issue.